### PR TITLE
[SMALLFIX] Update S3A directory suffix

### PR DESCRIPTION
**Pull request by madanadit**
Update the suffix for pseudo-directories in S3A to path separator.

Caveats when migrating from older versions:
- 1.4.0 directory breadcrumbs will show up as files
- `isDirectory` will not recognize 1.4.0 empty directories
- `deleteDirectory` will not delete 1.4.0 empty directories. Also performance could be impacted as `UfsSyncChecker` may falsely detect nested directory breadcrumbs as un-synced files

### CODE REVIEW

**Comment by madanadit** [comment]
@calvinjia Please take a look. Thanks!

File: underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java [hunk 1]
```diff
@@ -68,7 +68,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(S3AUnderFileSystem.class);
 
   /** Suffix for an empty file to flag it as a directory. */
-  private static final String FOLDER_SUFFIX = "_$folder$";
+  private static final String FOLDER_SUFFIX = "/";
```
Comment (calvinjia) [inline underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java]: Could we make this configurable?

**Comment by madanadit** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
@madanadit Could you run the s3 test profile with s3a and ensure those pass?

**Comment by madanadit** [comment]
@gpang Can you also take a look?

**Comment by gpang** [comment]
@madanadit Is there a restriction that directory names cannot end with the suffix?

**Comment by madanadit** [comment]
@gpang there is no such restriction

**Comment by madanadit** [comment]
@calvinjia Everything that ran passed. A couple timed out (seems unrelated to this PR)
```
FileInStreamIntegrationTest.remoteReadLargeFile
FileSystemMasterIntegrationTest.createFilePerf:295
FileSystemMasterIntegrationTest.listFiles:489
FileSystemMasterIntegrationTest.listStatus:515
```

**Pull request merged**